File: python/solved/problem_01.py
def summation(k: int, n: int) -> int:
    p = (n - 1) // k
    return k * p * (p + 1) // 2


def multiples(number: int) -> int:
    return summation(3, number) + summation(5, number) - summation(3 * 5, number)


if __name__ == "__main__":
    m = multiples(1000)
    print(m)

File: haskell/solved/problem_01.hs
summation :: Integer -> Integer -> Integer
summation k n =
    k * p * (p + 1) `div` 2 where
        p = (n - 1) `div` k


multiples :: Integer -> Integer
multiples number =
    summation 3 number +
    summation 5 number -
    summation 15 number


main = do
    print(multiples limit)
    where
        limit = 1000